web3/_utils/contracts.py

```diff
@@ -210,43 +208,6 @@ def find_matching_fn_abi(
         raise Web3ValidationError(message)
 
 
-def encode_abi(
-    w3: Union["AsyncWeb3", "Web3"],
-    abi: ABIFunction,
-    arguments: Sequence[Any],
-    data: Optional[HexStr] = None,
-) -> HexStr:
-    argument_types = get_abi_input_types(abi)
-
-    if not check_if_arguments_can_be_encoded(abi, w3.codec, arguments, {}):
-        raise Web3TypeError(
-            "One or more arguments could not be encoded to the necessary "
-            f"ABI type. Expected types are: {', '.join(argument_types)}"
-        )
-
-    normalizers = [
-        abi_address_to_hex,
-        abi_bytes_to_bytes,
-        abi_string_to_text,
-    ]
-    if not w3.eth.is_async:
```

Comment:
See my comment about removing the `is_async` from the utility method. I think we can just build the appropriate normalizers here in the code block instead of trying to use an `is_async` flag for a general public utility method that's trying to encode abi. We can build the right normalizers for each code block and pass it in instead, one appropriate for the sync code block and another that is appropriate for the async code.